packages/client/src/objectSet/ObjectSetListenerWebsocket.ts

```diff
@@ -173,31 +177,49 @@ export class ObjectSetListenerWebsocket {
       globalThis.crypto ??= (await import("node:crypto")).webcrypto as any;
     }
 
-    const objDef = await this.#client.ontologyProvider.getObjectDefinition(
-      objectType.apiName,
-    );
+    const objDef = objectType.type === "object"
+      ? await this.#client.ontologyProvider.getObjectDefinition(
+        objectType.apiName,
+      )
+      : await this.#client.ontologyProvider.getInterfaceDefinition(
+        objectType.apiName,
+      );
 
-    if (properties.length === 0) {
-      properties = Object.keys(objDef.properties) as Array<P>;
-    }
+    let objectProperties: Array<P> = [];
+    let referenceProperties: Array<P> = [];
 
-    const objectProperties = properties.filter((p) =>
-      objDef.properties[p].type !== "geotimeSeriesReference"
-    );
-    const referenceProperties = properties.filter((p) =>
-      objDef.properties[p].type === "geotimeSeriesReference"
-    );
+    if (objectType.type === "object") {
+      if (properties.length === 0) {
+        properties = Object.keys(objDef.properties) as Array<P>;
+      }
+
+      objectProperties = properties.filter((p) =>
+        objDef.properties[p].type !== "geotimeSeriesReference"
+      );
+
+      referenceProperties = properties.filter((p) =>
+        objDef.properties[p].type === "geotimeSeriesReference"
+      );
+    } else {
+      objectProperties = [];
```

Comment:
OSW assumes the names we provide them are the API names for the actual object type implementations instead of the SPT names. Therefore the current API I decided on was assuming if you pass an interface in that you are only specifying reference properties. This does require any casting what is passed in, and I think it's a larger conversation if the TS OSDK supports specifying anything beyond an SPT.

Reply:
There are ongoing discussions to allow OSW to take in the SPT names instead of the object type property API names here. I'll also open up the discussion as to how we want to allow requesting reference properties in the finalized API